### PR TITLE
Feature/implement dao for retrieving folder versions

### PR DESCRIPTION
**Pull request by lmarcoruiz**
Implements functionality and tests with mock data for retrieving Folders based on a timestamp.

### CODE REVIEW

**Comment by axelsiebert** [comment]
fixes #52

**Comment by chevalleyc** [comment]
Spotted issue with PG version
```
[ERROR] /home/circleci/project/service/src/test/java/org/ehrbase/dao/access/jooq/FolderAccessHistoryMockDataProvider.java:[56,57] cannot find symbol
  symbol:   variable POSTGRES_9_5
  location: class org.jooq.SQLDialect
[ERROR] /home/circleci/project/service/src/test/java/org/ehrbase/dao/access/jooq/FolderAccessHistoryTest.java:[84,48] cannot find symbol
  symbol:   variable POSTGRES_9_5
  location: class org.jooq.SQLDialect
```

File: service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccessHistory.java [hunk 1]
```diff
@@ -0,0 +1,667 @@
...
+            result2 = domainAccess.getContext().select(count().as("countVersionInTable")).from(FOLDER).where(FOLDER_HISTORY.SYS_TRANSACTION.lessOrEqual(timeCommitted).and(FOLDER_HISTORY.ID.eq(vFolderUid))).fetchOne();
+        } catch (RuntimeException e) { // generalize SQL exceptions
+            throw new InternalServerException("Problem with SQL statement or input", e);
+        }
+        int versionComHist = (int) result2.get("countVersionInTable");
+        if (timeCommitted.compareTo(latestCompoTime) >= 0) {//if the timestamp is after or equal to the sys_transaction of the latest folder available, add one since its version has not been counted for being the one stored in the ehr.folder table
+            versionComHist++;
+        }
+        if (versionComHist == 0) {
+            throw new IllegalArgumentException("There are no versions available prior to date " + timeCommitted + " for the the FOLDER with id: " + vFolderUid);
```
[the remark this answers is not in the document]
Comment (lmarcoruiz) [reply service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccessHistory.java]: done

File: service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccessHistory.java [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+import org.postgresql.util.PGobject;
+import java.sql.Timestamp;
+import java.util.*;
+import static org.ehrbase.jooq.pg.Tables.*;
+import static org.jooq.impl.DSL.*;
+
+/***
+ *@Created by Luis Marco-Ruiz on Jun 13, 2019
+ */
+public class FolderAccessHistory extends DataAccess implements I_FolderAccess, Comparable<FolderAccessHistory> {
```
[the remark this answers is not in the document]
Comment (lmarcoruiz) [reply service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccessHistory.java]: done

**Comment by axelsiebert** [comment]
> Spotted issue with PG version
> 
> ```
> [ERROR] /home/circleci/project/service/src/test/java/org/ehrbase/dao/access/jooq/FolderAccessHistoryMockDataProvider.java:[56,57] cannot find symbol
>   symbol:   variable POSTGRES_9_5
>   location: class org.jooq.SQLDialect
> [ERROR] /home/circleci/project/service/src/test/java/org/ehrbase/dao/access/jooq/FolderAccessHistoryTest.java:[84,48] cannot find symbol
>   symbol:   variable POSTGRES_9_5
>   location: class org.jooq.SQLDialect
> ```

I think this an issue raised by migration and dependency updates. The current version of PG is 11.5 and maybe the dialect follows this version.

**Pull request merged**